src/components/controls/TextArea/hoc/withCounter.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+import {Text} from '../../../Text';
+import {getComponentName} from '../../../utils/getComponentName';
+import {TextArea, TextAreaProps} from '../TextArea';
+
+export interface WithCounterProps {
+    maxLength: number;
```

Comment:
This name is looks like something related to input width

I guess something like `counterMax` would be better